core/src/org/pentaho/di/core/extension/ExtensionPointMap.java

```diff
@@ -90,7 +90,10 @@ public static ExtensionPointMap getInstance() {
 
   /**
    * Add the extension point plugin to the map
-   * 
+   *
+   * Warning: Do not call this method directly through ExtensionPointMap, it may lead to concurrency problems. Call
+   * PluginRegistry.getInstance().registerPlugin() instead.
+   *
    * @param extensionPointPlugin
    */
   public void addExtensionPoint( PluginInterface extensionPointPlugin ) {
```

Comment:
Shouldn't we change the visibility of this method to make it visible inside the package only? 

Or if the class is not to be used from outside, maybe it should be a private class in PluginRegistry?